x-pack/plugins/ingest_pipelines/public/application/components/pipeline_editor/__jest__/processors/append.test.tsx

```diff
@@ -0,0 +1,128 @@
+import { setup, SetupResult, getProcessorValue } from './processor.helpers';
+
+// Default parameter values automatically added to the URI parts processor when saved
+const defaultAppendParameters = {
+  if: undefined,
+  tag: undefined,
+  ignore_failure: undefined,
+  description: undefined,
+};
+
+const APPEND_TYPE = 'append';
+
+describe('Processor: Append', () => {
+  let onUpdate: jest.Mock;
+  let testBed: SetupResult;
+
+  beforeAll(() => {
+    jest.useFakeTimers();
+  });
+
+  afterAll(() => {
+    jest.useRealTimers();
+  });
+
+  beforeEach(async () => {
+    onUpdate = jest.fn();
+
+    await act(async () => {
+      testBed = await setup({
+        value: {
+          processors: [],
+        },
+        onFlyoutOpen: jest.fn(),
+        onUpdate,
+      });
+    });
+    testBed.component.update();
+  });
+
+  test('prevents form submission if required fields are not provided', async () => {
+    const {
+      actions: { addProcessor, saveNewProcessor, addProcessorType },
+      form,
+    } = testBed;
+
+    // Open flyout to add new processor
+    addProcessor();
+
+    // Add type (the other fields are not visible until a type is selected)
+    await addProcessorType(APPEND_TYPE);
+
+    // Click submit button with only the type defined
+    await saveNewProcessor();
+
+    // Expect form error as "field" is required parameter
+    expect(form.getErrorsMessages()).toEqual([
+      'A field value is required.',
+      'A value is required.',
+    ]);
+  });
+
+  test('saves with default parameter values', async () => {
+    const {
+      actions: { addProcessor, saveNewProcessor, addProcessorType },
+      form,
+      find,
+    } = testBed;
+
+    // Open flyout to add new processor
+    addProcessor();
+    // Add type (the other fields are not visible until a type is selected)
+    await addProcessorType(APPEND_TYPE);
+    // Add "field" value (required)
+    form.setInputValue('fieldNameField.input', 'field_1');
+
+    find('valueFieldAppend.input').simulate('change', [{ label: 'Some_Value' }]);
+
+    // Save the field
+    await saveNewProcessor();
+
+    const processors = getProcessorValue(onUpdate, APPEND_TYPE);
+    expect(processors[0].append).toEqual({
+      field: 'field_1',
+      value: ['Some_Value'],
+      ...defaultAppendParameters,
+    });
+  });
+
+  test('allows optional parameters to be set', async () => {
+    const {
+      actions: { addProcessor, addProcessorType, saveNewProcessor },
+      form,
+      find,
+    } = testBed;
+
+    // Open flyout to add new processor
+    addProcessor();
+    // Add type (the other fields are not visible until a type is selected)
+    await addProcessorType(APPEND_TYPE);
+    // Add "field" value (required)
+    form.setInputValue('fieldNameField.input', 'field_1');
+
+    // Set optional parameteres
+    find('valueFieldAppend.input').simulate('change', [{ label: 'Some_Value' }]);
+
+    form.toggleEuiSwitch('ignoreFailureSwitch.input');
```

Comment:
The ignore failure switch is common across processors. Similar to https://github.com/elastic/kibana/pull/95130#discussion_r600489090, could we add a single test that covers the common optional parameters? (Off the top of my head, I think this includes `ignore_failure`, `tag`, and `condition`).

Reply:
We could do this but which processor would we do it with? We can't save a processor without picking a type which means that this test would need to be included with one of the processors. I can just include it with the append processor since that is the first one in the list and remove the common fields from all other tests. Does that make sense? @alisonelizabeth

Reply:
Good point. Maybe we could create a helper that we could use in each test that performed the common field actions? At a minimum for this PR, I think we should at least add coverage for setting the `tag` and `condition` fields.

Reply:
Done.